audio-worklet/noisegate-test.html

```diff
@@ -0,0 +1,451 @@
+
+        // The signal should be all 1's prior to timeToThreshold to simplify
+        // the math that relates the closing index to the timeConstant.
+        let gain = new GainNode(context, {gain : 1});
+        let constantSource = new ConstantSourceNode(context, {offset: 1});
+        constantSource.connect(gain).connect(noisegate.input);
+        noisegate.output.connect(context.destination);
+
+        // The post-threshold gain should make samples less than the threshold.
+        // It must be greater than 0; otherwise samples will immediately
+        // be 0 after the threshold, and there will be no inspectable delay.
+        const beneathThresholdGain = 0.1;
+        gain.gain.setValueAtTime(beneathThresholdGain, timeToThreshold);
+        
+        constantSource.start();
+
+        // The noise gate output should contain four phases:
+        //   beforeOpenPhase (all 0) with a length determined by timeConstant.
+        //   openPhase (all 1) which ends at timeToThreshold.
+        //   betweenOpenAndClosePhase (all beneathThresholdGain) with a length
+        //       determined by timeConstant.
+        //   closePhase (all 0) for the remainder of the channel.
+        context.startRendering().then((buffer) => {
+          let channel = buffer.getChannelData(0);
+          
+          // With a constant source, the number of steps it takes for the
+          // envelope to open and close is a function of the time constant
+          // relative to the threshold.
+          const numOpeningFrames =
+              toIndex_(getTimeToTarget_(0, 1, timeConstant, linearThreshold),
+                       sampleRate);
+          
+          const numClosingFrames =
+              toIndex_(getTimeToTarget_(1, Math.pow(beneathThresholdGain, 2),
+                                        timeConstant, linearThreshold),
+                       sampleRate);
+
+          const openIndex = numOpeningFrames + scriptProcessorLatency;
+          
+          const sourceBeneathThresholdIndex =
+              testFrames * timeToThreshold + scriptProcessorLatency + 1;
+
+          let beforeOpenPhase =
+              channel.slice(scriptProcessorLatency, openIndex);
+          
+          let openPhase = channel.slice(openIndex, sourceBeneathThresholdIndex);
+
+          let betweenOpenAndClosePhase =
+              channel.slice(sourceBeneathThresholdIndex,
+                            sourceBeneathThresholdIndex + numClosingFrames);
+          
+          let closePhase = channel.slice(
+              sourceBeneathThresholdIndex + numClosingFrames);
+          
+          should(beforeOpenPhase, 'The samples before the gate opens')
+              .beConstantValueOf(0);
+          
+          should(openPhase, 'The open phase').beConstantValueOf(1);
+          
+          should(betweenOpenAndClosePhase, 'The samples before the gate closes')
+              .beCloseToArray(new Array(betweenOpenAndClosePhase.length)
```

Comment:
'beCloseToArray' or 'beEqualToArray'?  And if the array is constant, should it be 'beConstantValueOf(Math.fround(beneathThresholdGain))`?